### PR TITLE
module 8 servlets & JSP

### PR DESCRIPTION
**Pull request by alextimonov**

### CODE REVIEW

File: ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    public static final String ENABLED_TASK = "Enabled Task=";
+    public static final String DELETED_TASK = "Deleted Task=";
+
+    // adds to session new attribute from request, removes from session attributes marked in request to delete
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        HttpSession session = request.getSession();
+        String parameter = request.getParameter("taskName");
+        if (isValid(parameter)) {
```
Comment (schaffe) [inline ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java]: what if parameter is not valid?

File: ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            String attributeName = ENABLED_TASK + parameter;
+            String attributeObject = request.getParameter("categoryName");
```
Comment (schaffe) [inline ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java]: validate

File: ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            session.setAttribute(attributeName, attributeObject);
+        }
+
+        Enumeration<String> parameterNames = request.getParameterNames();
```
Comment (schaffe) [inline ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java]: use for-loop iteration or streams. Enumeration is to low level

File: ToDoList/src/main/resources/WEB-INF/web.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
+         version="3.1">
+
+    <display-name> Simple TODO List </display-name>
```
Comment (schaffe) [inline ToDoList/src/main/resources/WEB-INF/web.xml]: don't use whitespaces in xml

File: ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java (hunk 1 above)
Comment (alextimonov) [reply ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java]: This parameter is a task to add. If it's blank, it shouldn't be added to the task list.  It won't be added to session's attributes and won't be shown on a page.

File: ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java (hunk 2 above)
Comment (alextimonov) [reply ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java]: fixed - added validation for "categoryName"

File: ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java (hunk 3 above)
Comment (alextimonov) [reply ToDoList/src/main/java/ua/goit/timonov/enterprise/module_8/UpdateServlet.java]: But method getParameterNames() from javax.servlet.ServletRequest returns Enumeration<String>.
Fixed iteration as it suggested in Javadoc for java.util.Enumeration<E>:
`for (Enumeration<String> parameterNames = request.getParameterNames(); parameterNames.hasMoreElements(); ) {`

File: ToDoList/src/main/resources/WEB-INF/web.xml (hunk 4 above)
Comment (alextimonov) [reply ToDoList/src/main/resources/WEB-INF/web.xml]: Fixed

**Comment by schaffe** [comment]
To be merged

**Pull request merged**